source/Calamari/ArgoCD/Git/RepositoryWrapper.cs

```diff
@@ -87,41 +87,41 @@ public void StageFiles(string[] filesToStage)
         public async Task PushChanges(bool requiresPullRequest,
                                       string summary,
                                       string description,
-                                      GitBranchName branchName,
+                                      GitReference branchName,
                                       CancellationToken cancellationToken)
         {
             var currentBranchName = repository.GetBranchName(branchName);
             var pushToBranchName = currentBranchName;
             if (requiresPullRequest)
             {
-                pushToBranchName = CalculateBranchName();
+                pushToBranchName = new GitBranchName(CalculateBranchName());
```

Comment:
nit: If this used the GitBranchName.FromFriendlyName(calc()). -then the prefix doens't ned to escape the GitBranchname class.

Reply:
👍 i might take a look at this in one of the higher up PRs to avoid rebasing it all again.